pkg/controller/deployment/generic_deployment_controller.go

```diff
@@ -80,7 +92,14 @@ func (c *genericDeploymentController) Sync(ctx context.Context, syncContext fact
 	for index := range deployment.Spec.Template.Spec.Containers {
 		deployment.Spec.Template.Spec.Containers[index].Image = certManagerImage(deployment.Spec.Template.Spec.Containers[index].Image)
 	}
-	_, opStatus, _, _ := c.operatorClient.GetOperatorState()
+	operatorSpec, opStatus, _, _ := c.operatorClient.GetOperatorState()
+	unsupportedExtensions, err := operatorclient.GetUnsupportedConfigOverrides(operatorSpec)
+	if err != nil {
+		return nil, false, append(errors, fmt.Errorf("Getting unsupportedConfigOverrides failed: %w", err))
+	}
+
+	deployment = c.customOverridesFunc(deployment, unsupportedExtensions)
```

Comment:
unless this is dictated by library-go, why would we need to the abstraction via a custom overrides func, why not executing it directly in the Sync method? The function could still be tested in isolation.

Reply:
aaaahh.. it's a leftover from previous approach. Good catch, let me refactor it.

Reply:
Fixed.